src/main/java/au/id/micolous/metrodroid/transit/podorozhnik/PodorozhnikTransitData.java

```diff
@@ -89,20 +112,117 @@ public String getCardName() {
     @Override
     public void writeToParcel(Parcel dest, int i) {
         dest.writeInt(mBalance);
+	    dest.writeInt(mLastTopup);
+	    dest.writeInt(mLastTopupTime);
+	    dest.writeInt(mLastSpend);
+	    dest.writeInt(mLastSpendTime);
+	    dest.writeInt(mLastTripTime);
+	    dest.writeInt(mLastValidator);
+	    dest.writeInt(mGroundCounter);
+	    dest.writeInt(mSubwayCounter);
     }
 
     @SuppressWarnings("UnusedDeclaration")
     public PodorozhnikTransitData(Parcel p) {
         mBalance = p.readInt();
+    	mLastTopup = p.readInt();
+	    mLastTopupTime = p.readInt();
+	    mLastSpend = p.readInt();
+	    mLastSpendTime = p.readInt();
+	    mLastTripTime = p.readInt();
+	    mLastValidator = p.readInt();
+	    mGroundCounter = p.readInt();
+	    mSubwayCounter = p.readInt();
     }
 
     public static TransitIdentity parseTransitIdentity(ClassicCard card) {
         return new TransitIdentity(NAME, null);
     }
 
-    public PodorozhnikTransitData(ClassicCard card) {
+    private void decodeSector4(ClassicCard card) {
         ClassicSector sector4 = card.getSector(4);
-        mBalance = getBalance(sector4);
+
+	    if (sector4 instanceof UnauthorizedClassicSector)
+	        return;
+
+	    // Block 0 and block 1 are copies. Let's use block 0
+	    byte[] block0 = sector4.getBlock(0).getData();
+	    byte[] block2 = sector4.getBlock(2).getData();
+	    byte[] b;
+	    b = Utils.reverseBuffer(block0, 0, 4);
+        mBalance = Utils.byteArrayToInt(b, 0, 4);
+	    b = Utils.reverseBuffer(block2, 8, 3);
+	    mLastTopup = Utils.byteArrayToInt(b, 0, 3);
+	    b = Utils.reverseBuffer(block2, 2, 3);
+	    mLastTopupTime = Utils.byteArrayToInt(b, 0, 3);
+    }	
+
+    private void decodeSector5(ClassicCard card) {
+        ClassicSector sector5 = card.getSector(5);
+
+	    if (sector5 instanceof UnauthorizedClassicSector)
+	        return;
+
+	    // Block 1 and block 2 are copies. Let's use block 2
+	    byte[] block0 = sector5.getBlock(0).getData();
+	    byte[] block1 = sector5.getBlock(1).getData();
+	    byte[] b;
+
+	    b = Utils.reverseBuffer(block0, 6, 4);
+	    mLastSpend = Utils.byteArrayToInt(b, 0, 4);
+	    b = Utils.reverseBuffer(block0, 0, 3);
+	    mLastSpendTime = Utils.byteArrayToInt(b, 0, 3);
+	    b = Utils.reverseBuffer(block0, 4, 2);
+	    mLastValidator = Utils.byteArrayToInt(b, 0, 2);
+	    b = Utils.reverseBuffer(block1, 2, 3);
+	    mLastTripTime = Utils.byteArrayToInt(b, 0, 3);
+	    mGroundCounter = block1[0] & 0xff;
+	    mSubwayCounter = block1[1] & 0xff;
+    }	
+
+    public PodorozhnikTransitData(ClassicCard card) {
+	    decodeSector4(card);
+	    decodeSector5(card);
+    }
+
+    private static Calendar convertDate(int mins) {
+        GregorianCalendar g = new GregorianCalendar(TZ);
+        g.setTimeInMillis(PODOROZHNIK_EPOCH);
+        g.add(GregorianCalendar.MINUTE, mins);
+        return g;
+    }
+
+    @Override
+    public List<ListItem> getInfo() {
```

Comment:
Even though there is only 1 trip/top-up available, please use the standard travel history interface for this stuff.